src/convolution.jl

```diff
@@ -0,0 +1,132 @@
+export ConvolutionInterpolation, ConvolutionMethod
+
+struct ConvolutionKernel end
+
+struct ConvolutionMethod <: InterpolationType end
+
+function (::ConvolutionKernel)(s)
+    s_abs = abs(s)
+    if s_abs < 1.0
+        return (3/2)*s_abs^3 - (5/2)*s_abs^2 + 1
+    elseif s_abs < 2.0
+        return -(1/2)*s_abs^3 + (5/2)*s_abs^2 - 4*s_abs + 2
+    else
+        return 0.0
+    end
+end
+
+struct ConvolutionInterpolation{T,N,TCoefs<:AbstractArray,IT<:NTuple{N,ConvolutionMethod},Axs<:Tuple} <: AbstractInterpolation{T,N,IT}
+    coefs::TCoefs
+    knots::Axs
+    it::IT
+    h::NTuple{N,Float64}
+    kernel::ConvolutionKernel
+end
+
+function ConvolutionInterpolation(knots::NTuple{N,AbstractVector}, vs::AbstractArray{T,N}) where {T,N}
+    if N > 2
+        error("ConvolutionInterpolation is currently implemented only for 1D and 2D. Got $N dimensions.")
+    end
+
+    coefs = create_coefs(knots, vs)
+    h = map(k -> k[2] - k[1], knots)
+    it = ntuple(_ -> ConvolutionMethod(), N)
+    if N == 1
+        knots_new = (knots[1][1]-h[1]:h[1]:knots[1][end]+h[1],)
+    else
+        knots_new = (knots[1][1]-h[1]:h[1]:knots[1][end]+h[1], knots[2][1]-h[2]:h[2]:knots[2][end]+h[2])
+    end
+
+    ConvolutionInterpolation{T,N,typeof(coefs),typeof(it),typeof(knots)}(coefs, (knots_new), it, h, ConvolutionKernel())
+end
+
+function create_coefs(knots::Tuple{AbstractVector}, vs::AbstractVector{T}) where T
```

Comment:
This name seems too general. Perhaps rename is to `create_cubic_convolutional_coefs`?

Reply:
Makes sense, done.